### PR TITLE
*: Support insert ignore

### LINKED ISSUE #1570: Support  insert ignore into
See https://dev.mysql.com/doc/refman/5.7/en/insert.html

### PR DESCRIPTION
**Pull request by shenli**
Ignore duplicate key errors for insert ignore stmt
Fix #1570
@coocood @zimulala @tiancaiamao @hanfei1991 PTAL

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -455,6 +455,27 @@ func (s *testSuite) TestInsertAutoInc(c *C) {
...
+
+	r := tk.MustQuery("select * from t;")
+	rowStr := fmt.Sprintf("%v %v", "1", "2")
+	r.Check(testkit.Rows(rowStr))
+
+	tk.MustExec("insert ignore into t values (1, 3)")
+
+	r = tk.MustQuery("select * from t;")
+	rowStr = fmt.Sprintf("%v %v", "1", "2")
+	r.Check(testkit.Rows(rowStr))
```
Comment (tiancaiamao) [inline executor/executor_test.go]: what about the result "row affected" on this change?

Comment (shenli) [inline executor/executor_test.go]: TiDB behave the same way with MySQL now.
MySQL:
    mysql> insert ignore into t values (1, 3);
    Query OK, 0 rows affected (0.00 sec)
TiDB:
mysql> insert ignore into t values (1, 3);
Query OK, 0 rows affected (0.00 sec)

Comment (tiancaiamao) [inline executor/executor_test.go]: LGTM

File: executor/executor_write.go [hunk 2]
```diff
@@ -470,6 +471,12 @@ func (e *InsertExec) Next() (*Row, error) {
 		}
 
 		if len(e.OnDuplicate) == 0 || !terror.ErrorEqual(err, kv.ErrKeyExists) {
+			// If you use the IGNORE keyword, errors that occur while executing the INSERT statement are ignored.
+			// For example, without IGNORE, a row that duplicates an existing UNIQUE index or PRIMARY KEY value in
+			// the table causes a duplicate-key error and the statement is aborted. With IGNORE, the row is discarded and no error occurs.
+			if e.Ignore {
+				return nil, nil
```
Comment (zimulala) [inline executor/executor_write.go]: Need to consider multi-row.

File: executor/executor_test.go [hunk 3]
```diff
@@ -455,6 +455,27 @@ func (s *testSuite) TestInsertAutoInc(c *C) {
...
+    create table t (id int PRIMARY KEY AUTO_INCREMENT, c1 int);`
+	tk.MustExec(testSQL)
+	testSQL = `insert into t values (1, 2);`
+	tk.MustExec(testSQL)
+
+	r := tk.MustQuery("select * from t;")
+	rowStr := fmt.Sprintf("%v %v", "1", "2")
+	r.Check(testkit.Rows(rowStr))
+
+	tk.MustExec("insert ignore into t values (1, 3)")
```
Comment (zimulala) [inline executor/executor_test.go]: Add a test about multi-row.

**Comment by shenli** [comment]
@zimulala PTAL

File: executor/executor_test.go [hunk 4]
```diff
@@ -455,6 +455,28 @@ func (s *testSuite) TestInsertAutoInc(c *C) {
...
+    create table t (id int PRIMARY KEY AUTO_INCREMENT, c1 int);`
+	tk.MustExec(testSQL)
+	testSQL = `insert into t values (1, 2);`
+	tk.MustExec(testSQL)
+
+	r := tk.MustQuery("select * from t;")
+	rowStr := fmt.Sprintf("%v %v", "1", "2")
+	r.Check(testkit.Rows(rowStr))
+
+	tk.MustExec("insert ignore into t values (1, 3), (2, 3)")
```
Comment (zimulala) [inline executor/executor_test.go]: Does it affect last_insert_id() ?

**Comment by tiancaiamao** [comment]
LGTM

File: executor/executor_test.go (hunk 4 above)
Comment (shenli) [inline executor/executor_test.go]: TiDB & MySQL have the same behavior:
mysql> create table t (id int PRIMARY KEY AUTO_INCREMENT, c1 int unique);
Query OK, 0 rows affected (0.01 sec)

mysql> insert ignore into t (c1) values (1), (2);
Query OK, 2 rows affected (0.01 sec)

mysql> select last_insert_id();
+------------------+
| last_insert_id() |
+------------------+
|                1 |
+------------------+
1 row in set (0.00 sec)

mysql> insert ignore into t values (4, 2);
Query OK, 0 rows affected (0.00 sec)

mysql> select last_insert_id();
+------------------+
| last_insert_id() |
+------------------+
|                1 |
+------------------+
1 row in set (0.00 sec)

**Comment by shenli** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**